### PR TITLE
Extend data pipeline to use real historical data

### PR DESCRIPTION
**Pull request by ghost**
Extends the data pipeline to real historical data and adds some script on how to actually get historical data from a per block frequency from cLabs BigQuery `eksportisto` and `cp_doto` data. 

- Data files are using [Github LFS](https://git-lfs.github.com/)
- Introduces a new simulation config parameter `DATA_SOURCE = 'historical'   # 'mock' or 'historical'`, where you indicate whether you want to use real historical data or mock data if using the `SCENARIO` mode in the MarketPriceGenerator. 
- To test the changes and use the scenario mode change this line in the [system parameters](https://github.com/celo-org/mento2-model/blob/main/model/system_parameters.py#L50) from `    model: List[MarketPriceModel] = default([MarketPriceModel.GBM])` to `    model: List[MarketPriceModel] = default([MarketPriceModel.SCENARIO])`

### CODE REVIEW

File: experiments/simulation_configuration.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
 BLOCKS_PER_TIMESTEP = 1  # number of blocks per timestep (=1 if sim on per-block-basis)
-SIMULATION_TIME_DAYS = 3  # number of days
+SIMULATION_TIME_DAYS = 2  # number of days
 # number of simulation_configuration timesteps
 TIMESTEPS = SIMULATION_TIME_DAYS * blocks_per_day // BLOCKS_PER_TIMESTEP
 TIMESTEPS_PER_YEAR = blocks_per_year // BLOCKS_PER_TIMESTEP
-MONTE_CARLO_RUNS = 2  # number of runs
+MONTE_CARLO_RUNS = 1  # number of runs
+DATA_SOURCE = 'historical'   # 'mock' or 'historical'
```
[the remark this answers is not in the document]
Comment (tobikuhlmann) [reply experiments/simulation_configuration.py:12]: it doesn't, the data is only processed when initiated by the market generator here https://github.com/celo-org/mento2-model/blob/main/model/generators/markets.py#L200.

Not sure though if simulation configuration is the right place for this variable, where would you put it?

**Comment by sissnad** [comment]
If I try to run it I get the following error from `historical_values.py`:
```
KeyError: "None of ['snapped_at'] are in the columns"
```

**Comment by tobikuhlmann** [comment]
> If I try to run it I get the following error from `historical_values.py`:
> 
> ```
> KeyError: "None of ['snapped_at'] are in the columns"
> ```

Do you have these do data files locally when you check out my branch? https://github.com/celo-org/mento2-model/blob/tobi/extendDataPipeline/data/celo_price_cap_volume.csv and https://github.com/celo-org/mento2-model/blob/tobi/extendDataPipeline/data/cusd_price_cap_volume.csv

They were previously committed into this repo and I refactored them to be stored with Github LFS. Not sure how this affects downloading the data when checking out a branch / cloning this repo

**Comment by tobikuhlmann** [comment]
you have to do it manually. If you install git lfs you can use `git lfs fetch --all` I think

**Comment by sissnad** [comment]
I have downloaded the files manually. Now `calculate_log_returns()` throws an error:
```
TypeError: unsupported operand type(s) for /: 'str' and 'str'
```

**Comment by tobikuhlmann** [comment]
uff haven't seen it before, for me all config switches worked. Can you see why it's thrown? And with what config?

**Comment by sissnad** [comment]
> uff haven't seen it before, for me all config switches worked. Can you see why it's thrown? And with what config?

Used your branch and changed the params:
```model: List[MarketPriceModel] = default([MarketPriceModel.SCENARIO])```

**Comment by tobikuhlmann** [comment]
can't reproduce ): 

- In `simulation_configuration.py`, do you have `mock` or `historical` for `DATA_SOURCE`?
- Does your `data/cusd_celo_example.csv` look like 

celo_usd,cusd_usd
2.25502326128682,0.9989
2.25502326128682,0.9989
2.25502326128682,0.9989
?

**Comment by sissnad** [comment]
> Does your data/cusd_celo_example.csv look like

For me the path is `data/historical_market_data/cusd_celo_example.csv`

And the following change fixes the problem for me; parenthesis around the relative returns

```python
def calculate_log_returns(self, data_frame):
        """
        calculates log returns out of a data frame with price time series in its columns
        """
        return data_frame.apply(lambda column: np.log((column/column.shift(1)).dropna()))
```

**Comment by tobikuhlmann** [comment]
still doesn't understand what the problem was. Anyways it doesn't change the calculation or result so let's do it this way, it's pushed

**Comment by sissnad** [comment]
> still doesn't understand what the problem was. Anyways it doesn't change the calculation or result so let's do it this way, it's pushed

`column/column.shift(1)` first row is Nan by construction. `column/column.shift(1).dropna()` first row is still NaB, because `dropna()`is only applied to the shifted column.  With parentheses  `(column/column.shift(1)).dropna()` the first row is removed

**Comment by sissnad** [comment]
Realising this I am surprised it does not fail for you 😅

**Comment by sissnad** [comment]
Looks good to me now!

**Review by sissnad — APPROVED** [review]

File: model/generators/markets.py [hunk 2]
```diff
@@ -197,13 +197,13 @@ def historical_returns(self, sample_size):
         historical log-returns"""
         # TODO Consider different sampling options
         # TODO Random Seed
-        data_feed = DataFeed(data_folder=DATA_FOLDER, data_file_name=DATA_FILE_NAME)
-        data = data_feed.historical_data
+        data_feed = DataFeed(data_folder=DATA_FOLDER)
+        data = data_feed.data
```
[the remark this answers is not in the document]
Comment (sissnad) [reply model/generators/markets.py:201]: Will you adapt this change?

**Comment by tobikuhlmann** [comment]
Now I am surprised too :D Indeed the first row remains NaN in the old code, but no error is thrown for me. Thanks for the catches!!

File: model/generators/markets.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobikuhlmann) [reply model/generators/markets.py]: yes I committed before merging

[the remark this answers is not in the document]
Comment (tobikuhlmann) [reply model/generators/markets.py]: dind't know about this before, good call out 👍

**Pull request merged**